### PR TITLE
addressing issues #15 #16 #17 #18 #19 #20

### PR DESCRIPTION
**Pull request by anewton1998**
This PR addresses the open issues #15 #16 #17 #18 #19 and #20

### CODE REVIEW

File: draft-regext-rdap-extensions.md [hunk 1]
```diff
@@ -94,7 +94,14 @@ For this reason, usage of an underscore character in RDAP extension identifiers
 is NOT RECOMMENDED. Implementers should be aware that many existing extension
 identifiers do contain underscore characters.
 
-# Usage in Queries
+[@!RFC7480] does not explicitly state that extension identifiers are case sensitive.
+This document updates the formulation in [@!RFC7480] to explicitly not that extension
```
Comment (jasdips) [inline draft-regext-rdap-extensions.md:98]: not -> note.

File: draft-regext-rdap-extensions.md [hunk 2]
```diff
@@ -181,16 +238,71 @@ The following example shows this use with a JSON object.
...
 is avoided by "lunarNIC_author".
 
+## Object Classes in Extensions {#object_classes_in_extensions}
+
+As described in [@!RFC9082] and #(usage_in_queries), an extension may define new paths in URIs.
+If the extension describes the behavior of an RDAP query using that path to return a new RDAP
+object classs, the JSON names are not required to be prepended with the extension identifier
+as described in (#child_json_values). However, the extension MUST define the value for the
+`objectClassName` string, and that value MUST be prepended with the extension identifier in
+the same manner as would be required for JSON names.
```
Comment (jasdips) [inline draft-regext-rdap-extensions.md:258]: That's interesting. Why?

File: draft-regext-rdap-extensions.md [hunk 3]
```diff
@@ -270,6 +387,18 @@ to technologies used by RDAP such as extended vCard/jCard properties.
...
+a `rel` attribute in a "links" array, where the `rel` value is registered in
+the Link Relations Registry (<https://www.iana.org/assignments/link-relations/link-relations.xhtml>).
+
+    "links": [
+      {
+        "value": "https://example.com/foo_bar",
+        "href" : "https://example.com/foo_baz",
+        "rel": "sideways"
+      }
+    ]
```
Comment (jasdips) [inline draft-regext-rdap-extensions.md:400]: Not clear from this example where the marker extension is leveraged. Is it "foo_baz" in "href"?

File: draft-regext-rdap-extensions.md [hunk 4]
```diff
@@ -469,7 +598,39 @@ processing of referrals, otherwise clients MUST assume the information
...
+3. Does the extension follow the JSON naming as described in (#usage_in_json)?
+4. If the extension is a newer version of an older extension, does
+the extension specification clearly describe if it is backwards compatible
+(see (#backwards_compatible_changes) or backwards incompatible
+(see (#backwards_incompatible_changes)).
+5. If the extension registers new values in an IANA registry used by RDAP,
+does it describe how a client is to use those values?
+
+Extension authors are encouraged but not required to seek an informal review
+of their extension by sending a request for review to regext@ietf.org.
```
Comment (jasdips) [inline draft-regext-rdap-extensions.md:631]: Like it.

File: draft-regext-rdap-extensions.md [hunk 5]
```diff
@@ -181,16 +239,74 @@ The following example shows this use with a JSON object.
...
+## Object Classes in Extensions {#object_classes_in_extensions}
+
+As described in [@!RFC9082] and #(usage_in_queries), an extension may define new paths in URIs.
+If the extension describes the behavior of an RDAP query using that path to return a new RDAP
+object classs, the JSON names are not required to be prepended with the extension identifier
+as described in (#child_json_values). However, the extension MUST define the value for the
+`objectClassName` string (which is used by clients to evaluate the type of the response), 
+and that value MUST be prepended with the extension identifier in
+the same manner as would be required for JSON names (to avoid collisions with object classes 
+defined in other extensions).
```
Comment (tomhrr) [inline draft-regext-rdap-extensions.md:261]: What about having objectClassName be the bare extension identifier?  That doesn't appear to pose any collision problems or similar.  (I don't have a strong opinion on this, but it seems like a reasonable thing to do that some people might find useful.)

File: draft-regext-rdap-extensions.md [hunk 6]
```diff
@@ -270,6 +391,30 @@ to technologies used by RDAP such as extended vCard/jCard properties.
...
+          "value": "https://example.com/domain/example.com",
+          "href" : "https://example.com/sideways_href",
+          "rel": "sideways",
+          "type": "application/rdap+json"
+        }
+      ]
+    }
+
+When defining the usage link relations, extensions SHOULD specify the
+media types expected to be used with those link relations.
```
Comment (tomhrr) [inline draft-regext-rdap-extensions.md:416]: SHOULD seems strong here, given that there could be any number of link relations usable with arbitrary media types.

File: draft-regext-rdap-extensions.md [hunk 7]
```diff
@@ -469,7 +614,39 @@ processing of referrals, otherwise clients MUST assume the information
...
+
+The RDAP Extensions Registry should have as a minimum three expert reviewers
+and ideally four or five. An expert reviewer assigned to the review of an RDAP
+extension registration must have another expert reviewer double check any
+submitted registration.
+
+Expert reviewers are to use the following criteria for extensions defined
+in this document, which include but are not limited to:
+
+1. Does the extension define an extension identifier follow the naming
```
Comment (tomhrr) [inline draft-regext-rdap-extensions.md:631]: following

**Comment by tomhrr** [comment]
All looks good, thanks.

**Pull request merged**